src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts

```diff
@@ -0,0 +1,594 @@
+
+			intrinsic: imageBasePath + 'intrinsic.png',
+
+			keyset: imageBasePath + 'keyset.png',
+
+			locate: imageBasePath + 'locate.png',
+
+			populationQuery: imageBasePath + 'population_query.png',
+
+			setFunction: imageBasePath + 'set_function.png',
+
+			snapshot: imageBasePath + 'snapshot.png',
+
+			spool: imageBasePath + 'spool.png',
+
+			tsql: imageBasePath + 'sql.png',
+
+			update: imageBasePath + 'update.png',
+
+			// fake operators
+			keyLookup: imageBasePath + 'bookmark_lookup.png',
+
+			// PDW operators
+			apply: imageBasePath + 'apply.png',
+
+			broadcast: imageBasePath + 'broadcast.png',
+
+			computeToControlNode: imageBasePath + 'compute_to_control_node.png',
+
+			constTableGet: imageBasePath + 'const_table_get.png',
+
+			controlToComputeNodes: imageBasePath + 'control_to_compute_nodes.png',
+
+			externalBroadcast: imageBasePath + 'external_broadcast.png',
+
+			externalExport: imageBasePath + 'external_export.png',
+
+			externalLocalStreaming: imageBasePath + 'external_local_streaming.png',
+
+			externalRoundRobin: imageBasePath + 'external_round_robin.png',
+
+			externalShuffle: imageBasePath + 'external_shuffle.png',
+
+			get: imageBasePath + 'get.png',
+
+			groupByApply: imageBasePath + 'apply.png',
+
+			groupByAggregate: imageBasePath + 'group_by_aggregate.png',
+
+			join: imageBasePath + 'join.png',
+
+			localCube: imageBasePath + 'intrinsic.png',
+
+			project: imageBasePath + 'project.png',
+
+			shuffle: imageBasePath + 'shuffle.png',
+
+			singleSourceRoundRobin: imageBasePath + 'single_source_round_robin.png',
+
+			singleSourceShuffle: imageBasePath + 'single_source_shuffle.png',
+
+			trim: imageBasePath + 'trim.png',
+
+			union: imageBasePath + 'union.png',
+
+			unionAll: imageBasePath + 'union_all.png'
+		};
+	}
+
+	private populate(node: azdata.QueryPlanGraphNode, diagramNode: any): any {
+
+		diagramNode.label = node.name;
+		if (node.properties && node.properties.length > 0) {
+			diagramNode.metrics = node.properties.map(e => { return { name: e.name, value: e.formattedValue.substring(0, 75) }; });
+		}
+
+		let icon = this._iconMap[node.type];
+		if (icon) {
+			diagramNode.icon = icon;
+		}
+
+		if (node.children) {
+			diagramNode.children = [];
+			for (let i = 0; i < node.children.length; ++i) {
+				diagramNode.children.push(this.populate(node.children[i], new Object()));
+			}
+		}
+		return diagramNode;
+	}
+
+	private createPlanDiagram(container: HTMLDivElement): void {
+		let diagramRoot: any = new Object();
+		let graphRoot: azdata.QueryPlanGraphNode = this._graph.root;
+		this.populate(graphRoot, diagramRoot);
+
+		new azdataGraph.azdataQueryPlan(container, diagramRoot, this._iconPaths);
+	}
+
+
+	public set graph(graph: azdata.QueryPlanGraph | undefined) {
+		this._graph = graph;
+		if (this._graph) {
+			/**
+			 * Create a show plan graph here.
+			 */
+			this._container.appendChild(document.createTextNode(`Query ${this._graphIndex}: `));
```

Comment:
it is weird to see this type of comment within a method. usually within a method, we use `//` styled comment.

Reply:
(also localize this text)